### PR TITLE
Refactor try catch in Management > Configuration

### LINKED ISSUE #3450: Refactor try catch in Management > Configuration
| Parent Issue |
|:---:|
|#3265| 

**Description**
In order to improve better error handling in the front-end, we should implement a better try-catch strategy in the `Management > Configuration` section, services, controller.

### PR DESCRIPTION
**Pull request by Machi3mfl**
Hi team,
This PR implement our new error handling strategy on the Ruleset section

Close #3450

### CODE REVIEW

File: public/controllers/management/components/management/configuration/configuration-switch.js [hunk 1]
```diff
@@ -99,12 +102,8 @@ class WzConfigurationSwitch extends Component {
   async componentDidMount() {
     // If agent, check if is synchronized or not
     if (this.props.agent.id !== '000') {
-      try {
-        const agentSynchronized = await agentIsSynchronized(this.props.agent);
-        this.setState({ agentSynchronized });
-      } catch (error) {
-        // do nothing
-      }
+      const agentSynchronized = await agentIsSynchronized(this.props.agent);
```
Comment (pablomarga) [inline public/controllers/management/components/management/configuration/configuration-switch.js:105]: Why you have deleted this try catch block?

File: public/controllers/management/components/management/configuration/utils/wz-fetch.js [hunk 2]
```diff
@@ -212,7 +212,7 @@ export const makePing = async (updateWazuhNotReadyYet, isCluster, tries = 30) =>
     }
     return Promise.resolve('Wazuh is ready');
   } catch (error) {
-    return Promise.reject('Wazuh could not be recovered.');
+    throw new error('Wazuh could not be recovered.');
```
Comment (Desvelao) [inline public/controllers/management/components/management/configuration/utils/wz-fetch.js:215]: **nitpick**: Fix the class constructor. Replace `error` with `Error`.

File: public/controllers/management/components/management/configuration/edit-configuration/edit-configuration.js [hunk 3]
```diff
@@ -96,42 +100,45 @@ class WzEditConfiguration extends Component {
...
-            <Fragment>
-              <EuiIcon type="alert" />
-              &nbsp;
-              <span>Error saving configuration</span>
-            </Fragment>
-          ),
-          color: 'danger',
-          text: typeof error === 'string' ? error : error.message
-        });
+        errorMessage = 'Error saving configuratio';
```
Comment (Desvelao) [inline public/controllers/management/components/management/configuration/edit-configuration/edit-configuration.js:107]: **nitpick**: Fix typo `configuratio` to `configuration`

File: public/controllers/management/components/management/configuration/utils/wz-fetch.js [hunk 4]
```diff
@@ -379,7 +379,7 @@ export const saveConfiguration = async (selectedNode, xml) => {
       await saveFileManager(xml);
     }
   } catch (error) {
-    return Promise.reject(error.message || error);
+    throw error;
```
Comment (Desvelao) [inline public/controllers/management/components/management/configuration/utils/wz-fetch.js:382]: **thought**: I am not sure if removing the throw of `error.message` could be a problem. I reviewed and it seems the `saveConfiguration` function is not used. Maybe we could remove it.

**Comment by matiasmoreno876** [comment]
LGTM!

File: public/controllers/management/components/management/configuration/configuration-switch.js (hunk 1 above)
Comment (Machi3mfl) [reply public/controllers/management/components/management/configuration/configuration-switch.js:105]: You're right. I revert this change.

File: public/controllers/management/components/management/configuration/utils/wz-fetch.js (hunk 4 above)
Comment (Machi3mfl) [reply public/controllers/management/components/management/configuration/utils/wz-fetch.js:382]: Thanks @Desvelao. I removed `saveConfiguration` from `wz-fetch` file

**Review by Desvelao — APPROVED** [review]
LGTM!

**Review by gabiwassan — APPROVED** [review]
LGTM!

**Pull request merged**